pkg/kubelet/kuberuntime/helpers.go

```diff
@@ -175,12 +176,23 @@ func isInitContainerFailed(status *kubecontainer.Status) bool {
 	return false
 }
 
-// GetStableKey generates a key (string) to uniquely identify a
-// (pod, container) tuple. The key should include the content of the
-// container, so that any change to the container generates a new key.
-func GetStableKey(pod *v1.Pod, container *v1.Container) string {
-	hash := strconv.FormatUint(kubecontainer.HashContainer(container), 16)
-	return fmt.Sprintf("%s_%s_%s_%s_%s", pod.Name, pod.Namespace, string(pod.UID), container.Name, hash)
+// GetBackoffKey generates a key (string) to uniquely identify a (pod, container) tuple for tracking
+// container backoff. The key should include any content of the container that is tied to the
+// backoff, so that any change generates a new key.
+func GetBackoffKey(pod *v1.Pod, container *v1.Container) string {
+	// Include stable identifiers (name, namespace, uid) as well as any
+	// fields that should reset the backoff when changed.
+	key := []string{
+		pod.Name,
+		pod.Namespace,
+		string(pod.UID),
+		container.Name,
+		container.Image,
+		container.Resources.String(),
```

Comment:
ah just thought of this, are these the container resources from the spec? 

Not sure if you care, but if so I think that will create a slight behavior change in IPPR, because today the backoff is reset only after the resize is successfully allocated. However if we just use the container resources from the spec to re-create the hash, wouldn't that essentially reset the backoff regardless of whether the resize is successfully allocated or not?

Reply:
Good callout, but we only ever call this function on the "allocated pod" (after the spec has been overwritten with the allocated resources): https://github.com/kubernetes/kubernetes/blob/42634e39d385b0dda8a96bc1be8807df94b2fbd2/pkg/kubelet/kuberuntime/kuberuntime_manager.go#L1547

Reply:
ahh got it ty for the explanation

Reply:
BTW, I was considering creating a new `type AllocatedPod Pod` that we could use to identify whether the pod had been updated to the allocated state, but the change blew up in size really quickly, so I never followed through.